pkg/apiserver/apiserver_test.go

```diff
@@ -785,6 +796,105 @@ func TestDelete(t *testing.T) {
 	}
 }
 
+func TestDeleteWithOptions(t *testing.T) {
+	storage := map[string]RESTStorage{}
+	simpleStorage := SimpleRESTStorage{}
+	ID := "id"
+	storage["simple"] = &simpleStorage
+	handler := handle(storage)
+	server := httptest.NewServer(handler)
+	defer server.Close()
+
+	grace := int64(300)
+	item := &api.DeleteOptions{
+		GracePeriod: &grace,
+	}
+	body, err := codec.Encode(item)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	client := http.Client{}
+	request, err := http.NewRequest("DELETE", server.URL+"/api/version/simple/"+ID, bytes.NewReader(body))
```

Comment:
If it works, JSON is consistent with all other operations and more friendly to our tooling and to forms.

Not surprisingly, others have run into the same issue:
http://stackoverflow.com/questions/299628/is-an-entity-body-allowed-for-an-http-delete-request
http://stackoverflow.com/questions/14323716/restful-alternatives-to-delete-request-body

Apparently it mostly works but there are known cases of it not working (e.g. Akamai).

Most common alternatives proposed are a special endpoint or query params. A subresource would be more consistent with our current direction.